streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -655,7 +661,7 @@ public StateStore getStateStore(final String name) {
      */
     @SuppressWarnings("WeakerAccess")
     public void close() {
-        if (task != null) {
+        if (task != null && !eosEnabled) {
```

Comment:
Why this? IMHO, we need to close the task, independent if EOS is enabled or not. However, we need to close the producer if EOS is not enabled
```
if (!eosEnabled) {
    producer.close();
}
```

Reply:
Fixed